src/domain/entities/TranslatableText.ts

```diff
@@ -16,3 +17,24 @@ export const buildTranslate = (locale: string): TranslateMethod => {
 };
 
 export type TranslateMethod = (string: TranslatableText) => string;
+
+export const importTranslate = <T extends TranslatableText>(item: T, language: string, term: string | undefined): T => {
+    if (term === undefined) {
+        return item;
+    } else if (language === "en") {
+        return { ...item, referenceValue: term };
+    } else {
+        return { ...item, translations: { ...item.translations, [language]: term } };
+    }
+};
+
+export const buildTranslationMap = (texts: TranslatableText[]) => {
```

Comment:
I would add a declarative return type to know what is returning (_also that return type could be used in ExtractTranslationsUseCase.ts#L8_), because it returns something like `Record<lang,Record<key,translatedText>` right?